core/pkg/fixhandler/fixhandler.go

```diff
@@ -368,3 +372,20 @@ func determineNewlineSeparator(contents string) string {
 		return unixNewline
 	}
 }
+
+// Handle the case where the resource file starts with ---
+// causes yaml.Node to misinterpret the resources
+func sanitizeYaml(fileAsString string) string {
+	if fileAsString[:3] == "---" {
+		fileAsString = "# " + fileAsString
+	}
+	return fileAsString
+}
+
+// For the --- case to ensure correct output file format
+func revertSanitizeYaml(fixedYamlString string) string {
+	if fixedYamlString[:5] == "# ---" {
```

Comment:
@XDRAGON2002 In case there are multiple resources (say, a Service, a Deployment, and a Pod), does this function handle that aspect? It should be generic enough to test on different [yamls.](https://raw.githubusercontent.com/kubernetes/website/main/content/en/examples/application/nginx-app.yaml)

// @vladklokun @craigbox @slashben

Reply:
@kiratsingh98 Yes it does, because the problem only occurs when the first line in the yaml is `---` so if the separator is used anywhere apart from that it's the same exact expected behavior as before.

As evident from the code changes as well, we're only concerned when the manifest begins with `---` everything else works fine.